packages/bsky/src/api/app/bsky/graph/getList.ts

```diff
@@ -68,9 +69,41 @@ const hydration = async (
       params.hydrateCtx,
     ),
   ])
+
+  const list = listState.lists?.get(listUri)
+  const ownerDid = new AtUri(listUri).hostname
+  if (
+    list?.record.purpose === 'app.bsky.graph.defs#referencelist' &&
+    // We show all users regardless of blocks if the viewer is the owner of the list, so no need to hydrate them
+    params.hydrateCtx.viewer !== ownerDid
+  ) {
+    const pairs = new Map()
+    for (const { did } of listitems) {
+      pairs.set(did, [did, ownerDid])
+    }
```

Comment:
If I'm not mistaken we'd be good with a single item `ownerDid -> listitems.map((li) => li.did)`.